### PR TITLE
Delete Queue

### PR DESCRIPTION
**Pull request by daltonfury42**
Delete queue impl

### CODE REVIEW

File: simplq/src/components/pages/Home/MyQueues.jsx [hunk 1]
```diff
@@ -2,20 +2,32 @@ import React from 'react';
...
-  if (myQueues.length === 0) {
-    return null;
-  }
+  const onDeleteClick = (queueId) => {
+    deleteQueue(queueId)
+      .then(() => getMyQueues().then((myNewQueues) => dispatch(setMyQueues(myNewQueues))))
+      .catch((err) => handleApiErrors(err))
+      .then(() => history.push('/'))
+      .catch((err) => handleApiErrors(err));
+  };
```
Comment (maaverik) [inline simplq/src/components/pages/Home/MyQueues.jsx:23]: Moving this to utilFns would be a good idea.

Comment (daltonfury42) [reply simplq/src/components/pages/Home/MyQueues.jsx:23]: In a PR, we can remove all `.catch((err) => handleApiErrors(err))` globally, and use a axios interceptor to handle this in one place.

Comment (maaverik) [reply simplq/src/components/pages/Home/MyQueues.jsx:23]: I actually meant moving the onDeleteClick fn since it's used in MyQueues and DeleteQueues

Comment (maaverik) [reply simplq/src/components/pages/Home/MyQueues.jsx:23]: As for the interceptor, I'll look at the error boundary and interceptor probably this weekend.

Comment (daltonfury42) [reply simplq/src/components/pages/Home/MyQueues.jsx:23]: Done. I have moved this into the queue service.

File: simplq/src/services/queue.js [hunk 2]
```diff
@@ -29,3 +30,10 @@ export const getStatusByName = async (queueName) => {
   const response = await makeRequest('get', `/queue/status?queueName=${queueName}`);
   return response.data;
 };
+
+export const deleteQueue = async (queueId) => {
+  const response = await makeRequest('delete', `/queue/${queueId}`);
+  await getMyQueues().then((myQueues) => store.dispatch(setMyQueues(myQueues)));
+  history.push('/');
+  return response.data;
```
Comment (maaverik) [inline simplq/src/services/queue.js:38]: Did you miss adding the catch or should I add exception handling while adding the axios interceptor?

**Review by maaverik — APPROVED** [review]

File: simplq/src/services/queue.js (hunk 2 above)
Comment (daltonfury42) [reply simplq/src/services/queue.js:38]: Let's add in one place... if you are planning to do that soon

**Pull request merged**